gulp-core-build-typescript/src/TypeScriptTask.ts

```diff
@@ -140,7 +146,14 @@ export class TypeScriptTask extends GulpTask<ITypeScriptTaskConfig> {
       .pipe(sourcemaps.init())
       .pipe(ts(tsProject, undefined, this.taskConfig.reporter));
 
-    allStreams.push(tsResult.js
+    // tslint:disable-next-line:typedef
+    const jsResult = (this.taskConfig.removeCommentsFromTypeScript
+      ? tsResult.js.pipe(require('gulp-decomment')({
+        space: true /* leave this on for sourcemaps */
```

Comment:
IIRC the sourcemaps are generated a compile time, but the sourcemaps go bad if we mess with the structure of the file after compilation, so instead of removal, we just replace comments with blanks. This ensures ts->js sourcemaps still work properly.